lib/git_worktree.rb

```diff
@@ -0,0 +1,336 @@
+require 'rugged'
+
+class GitWorktree
+  attr_accessor :name, :email, :base_name
+  ENTRY_KEYS = [:path, :dev, :ino, :mode, :gid, :uid, :ctime, :mtime]
+  DEFAULT_FILE_MODE = 0100644
+  LOCK_REFERENCE = 'refs/locks'
+  HEAD_REF = 'refs/heads/master'
+
+  def initialize(options = {})
+    raise ArgumentError, "Must specify path" unless options.key?(:path)
+    @path          = options[:path]
+    @email         = options[:email]
+    @name          = options[:name]
+    @bare          = options[:bare]
+    @commit_sha    = options[:commit_sha]
+    @password      = options[:password]
+    @fast_forward_merge = options[:ff] || true
+    @ssl_no_verify = options[:ssl_no_verify] || false
+    @remote_name   = 'origin'
+    @cred          = Rugged::Credentials::UserPassword.new(:username => @name,
+                                                           :password => @password)
+    @base_name     = File.basename(@path)
+    process_repo(options)
+  end
+
+  def delete_repo
+    return false unless @repo
+    @repo.close
+    FileUtils.rm_rf(@path)
+    true
+  end
+
+  def add(path, data, commit_sha = nil, default_entry_keys = {})
+    entry = {}
+    entry[:path] = path
+    ENTRY_KEYS.each { |key| entry[key] = default_entry_keys[key] if default_entry_keys.key?(key) }
+    entry[:oid]  = @repo.write(data, :blob)
+    entry[:mode] ||= DEFAULT_FILE_MODE
+    entry[:mtime] ||= Time.now
+    current_index(commit_sha).add(entry)
+  end
+
+  def remove(path, commit_sha = nil)
+    current_index(commit_sha).remove(path)
+  end
+
+  def remove_dir(path, commit_sha = nil)
+    current_index(commit_sha).remove_dir(path)
+  end
+
+  def file_exists?(path, commit_sha = nil)
+    find_entry(path, commit_sha) ? true : false
```

Comment:
If you *really* need a boolean, prefer `!!` over the ternary.  However, I question whether or not you really need the boolean.